### PR TITLE
Issue_638: Added fix for XML character in the command output.

### PR DESCRIPTION
**Pull request by geetanjalimanegslab**
# Please include a summary of the changes

* vane/tests_tools.py - Updated file with a utils to validate a XML character, 

# Any specific logic/part of code you need extra attention on
 

- Added utils to validate the XML characters in the command output
- Added check to format the command output first and then pass it for evidence section of docx report.

# Include the Issue number and link

Make sure to link the issue in the github PR UI 
#638 
# List/Attach any dependencies/past issues that are required for this change/provide context

# Type of change

- - [x] Bug fix (non-breaking change which fixes an issue)
- - [ ] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [ ] Other (please specify)

# Effort required on reviewers end

- - [ ] Easy
- - [x] Medium
- - [ ] Hard 

# How Has This Been Tested?

## Bug fix
  Note: Tested all the reports with hardcoded output.
   **Reports before adding the fix for the invalid characters in the command output**    
   
[Before_invalid_xml_character_fix.zip](https://github.com/user-attachments/files/15955954/Before_invalid_xml_character_fix.zip)

before_fix_xml_characters

   **Reports after adding the fix for invalid characters in the command output**    
     
[After_invalid_xml_character_fix.zip](https://github.com/user-attachments/files/15956209/After_invalid_xml_character_fix.zip)
 
image

## New feature

    Ensure current test cases pass and include newer test cases if required
    
# CI pipeline result

- - [ ] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure
  
# Verify Documentation Update

If applicable, ensure changes to [official documentation](http://vane.arista.com/) are included in the PR by following the steps below:

(1) You can locally make required documentation changes within the [docs](https://github.com/aristanetworks/vane/tree/develop/docs) folder and test them by following these steps:

- `mike deploy test`(or any name you want for the local changes version)
- `mike serve`
  (you will be able to view `test` as an option in the version dropdown)
- Once you have ensured the changes are accurate, execute `mike delete test`(this ensures you do not have any local  version branches which do not exist on gh-pages in the github repo).  

(2) Ensure you have [pre-commit hook](https://pre-commit.com/) installed which will take care of linting checks for documentation before you commit your changes.

(3) Lastly ensure the following jobs pass when you create the PR. These jobs ensure that the changes within your PR do not break the documentation.

- Documentation Testing / Run pre-commit validation hooks (pull_request) 
- Documentation Testing / Build site with no warnings (pull_request)  
   

    
# Additional comments

### CODE REVIEW

File: vane/tests_tools.py [hunk 1]
```diff
@@ -1792,3 +1795,35 @@ def add_cmds_evidence(self, cmds, cmds_output, device_name):
...
+        return (
+            0x20 <= codepoint <= 0xD7FF
+            or codepoint in (0x9, 0xA, 0xD)
+            or 0xE000 <= codepoint <= 0xFFFD
+            or 0x10000 <= codepoint <= 0x10FFFF
+        )
+
+    def get_cmd_output_from_device_details(self, command):
+        """
+        Utility to get the command output of the command mentioned in arguments.
```
Comment (hsisalgslab) [inline vane/tests_tools.py:1818]:
```suggestion
        Utility to get the output of the command mentioned in arguments.
```

File: vane/tests_tools.py [hunk 2]
```diff
@@ -1085,11 +1085,14 @@ def __init__(self, tests_definitions, test_suite, dut):
...
+                formatted_output = self.get_cmd_output_from_device_details(self.show_cmd)
+                self.show_cmd_txt = formatted_output
             for show_cmd in self.show_cmds[self.dut_name]:
-                self.show_cmd_txts[self.dut_name].append(self.dut["output"][show_cmd]["text"])
+                formatted_output = self.get_cmd_output_from_device_details(show_cmd)
+                self.show_cmd_txts[self.dut_name].append(formatted_output)
             for show_cmd in self._show_cmds[self.dut_name]:
-                self._show_cmd_txts[self.dut_name].append(self.dut["output"][show_cmd]["text"])
+                formatted_output = self.get_cmd_output_from_device_details(show_cmd)
+                self._show_cmd_txts[self.dut_name].append(formatted_output)
```
Comment (hsisalgslab) [inline vane/tests_tools.py:1095]: lines from 1093 to 1095 is repeated , Can we please confirm same ?

**Review by hsisalgslab — APPROVED** [review]

File: vane/tests_tools.py [hunk 3]
```diff
@@ -1792,3 +1795,35 @@ def add_cmds_evidence(self, cmds, cmds_output, device_name):
...
+        """
+        Utility to filter the string as per XML standard
+        Args:
+            char(ste): characters to filter out
+        Return:
+            str: Filtered string
+        """
+
+        codepoint = ord(char)
+        return (
```
Comment (kgrozis-arista) [inline vane/tests_tools.py:1809]: Can we walk thru logic?  I need to better understand why these ranges are being used.  I also want to understand the join expression better.

Comment (kgrozis-arista) [reply vane/tests_tools.py:1809]: I'm good with logic

**Review by kgrozis-arista — APPROVED** [review]

**Pull request merged**